### PR TITLE
K8SPG-72 document smart upgrade

### PR DESCRIPTION
**Pull request by fiowro**

### CODE REVIEW

File: source/update.rst [hunk 1]
```diff
@@ -37,6 +65,95 @@ The following steps will allow you to update both of them to current version
...
+
+Automatic upgrade
+*****************
+
+Starting from version 1.1.0, the Operator does fully automatic upgrades to
+the newer versions of Percona PostgreSQL Cluster within the method named *Smart
+Updates*.
+
+The Operator will carry on upgrades according to the following algorithm.
+It will query a special *Version Service* server at scheduled times to obtain
```
Comment (hors) [inline source/update.rst:83]: We need to mention that VS has status  'tech preview ' now and it is disabled by default in 1.1.0.
@spron-in  Am I right?

**Review by hors — CHANGES_REQUESTED** [review]

File: source/update.rst [hunk 2]
```diff
@@ -37,6 +65,98 @@ The following steps will allow you to update both of them to current version
...
+---------------------------------------------
+
+Automatic upgrade
+*****************
+
+Starting from version 1.1.0, the Operator does fully automatic upgrades to
+the newer versions of Percona PostgreSQL Cluster within the method named *Smart
+Updates*.
+
+.. note:: Smart Updates have technical preview status and are disabled by
```
Comment (hors) [inline source/update.rst:82]: Not a smart update but Version Service has technical preview status. And we disabled it by default (in CR), so  operator can't get new images from VS but if user sets a new images via CR and apply it the cluster will be updated in a smart way :)

**Review by hors — CHANGES_REQUESTED** [review]

**Review by hors — APPROVED** [review]

File: source/update.rst [hunk 3]
```diff
@@ -21,6 +28,27 @@ The following steps will allow you to update both of them to current version
...
 
+.. |rarr|   unicode:: U+02192 .. RIGHTWARDS ARROW
+
+#. If you upgrade the Operator **from a version earlier than 1.1.0**, the
+   following additional step is needed for the 1.0.0 |rarr| 1.1.0 upgrade.
+
+.. code:: bash
+
+   $ export CLUSTER=<cluster-name>
+   $ for user in postgres primaryuser $(kubectl get perconapgcluster/${CLUSTER} -o yaml | yq r - 'spec.user'); do args+="--from-literal=$user=$(kubectl get secret/${CLUSTER}-${user}-secret -o yaml | yq r - 'data.password' | base64 -d) "; done; eval kubectl create secret generic ${CLUSTER}-users-secret "${args}"
```
Comment (hors) [inline source/update.rst:39]:
```suggestion
   $ for user in postgres primaryuser $(kubectl get perconapgcluster/${CLUSTER} -o yaml | yq r - 'spec.user'); do args+="--from-literal=$user=$(kubectl get secret/${CLUSTER}-${user}-o yaml | yq r - 'data.password' | base64 -d) "; done; eval kubectl create secret generic ${CLUSTER}-users "${args}"
```
@cap1984  Am I right ?

Comment (cap1984) [reply source/update.rst:39]: yeap. It's true for default CR configuration

File: source/update.rst [hunk 4]
```diff
@@ -37,6 +65,100 @@ The following steps will allow you to update both of them to current version
...
+
+The following example sets the midnight update checks with the official
+Percona's Version Service:
+
+.. code:: yaml
+
+   spec:
+     upgradeOptions:
+       apply: recommended
+       versionServiceEndpoint: https://check-dev.percona.com
```
Comment (cap1984) [inline source/update.rst:139]: check.percona.com

**Review by cap1984 — APPROVED** [review]

**Pull request merged**